Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,342 @@
+---
+title: Implicitly-scoped qubit allocation
+description: Provides more convenient syntax for allocating qubits.
+author: Sarah Marshall
+date: September 28, 2020
+---
+
+# Proposal
+
+1. Add new `use` and `borrow` statements to allocate qubits that are valid until the end of the current block.
+2. Add new scope statement that does not require a preceding keyword, and is executed exactly once; equivalent to `if (true) { }`.
```

Comment:
I take it this is useful in this proposal for terminating the lifetime of an allocated qubit. Does it have uses beyond that?

Reply:
It limits the scope of any variable, including classical variables declared with `let` or `mutable`, see [Impact on Existing Mechanisms](https://github.com/microsoft/qsharp-language/blob/samarsha/qubit-allocation/Approved/implicitly-scoped-qubit-allocation.md#impact-on-existing-mechanisms).